File: 2020/src/day1.ts
import { Input } from "./input";

export const part1 = (input: Input) => {
  const numbers = input.asNumbers();

  let out;

  numbers.forEach((n1, i) =>
    numbers.slice(i).forEach((n2) => {
      if (n1 + n2 === 2020) {
        out = n1 * n2;
      }
    })
  );

  return out;
};

export const part2 = (input: Input) => {
  const numbers = input.asNumbers();

  let out;

  numbers.forEach((n1, i) =>
    numbers.slice(i).forEach((n2, j) => {
      numbers.slice(j).forEach((n3) => {
        if (n1 + n2 + n3 === 2020) {
          out = n1 * n2 * n3;
        }
      });
    })
  );

  return out;
};

File: 2017/day1.py
#!/usr/local/bin/python3


def part1(input_string):
    return sum(int(digit) for i, digit in enumerate(input_string) if digit == input_string[(i + 1) % len(input_string)])


def part2(input_string):
    return sum(
        int(digit)
        for i, digit in enumerate(input_string)
        if digit == input_string[(i + len(input_string) // 2) % len(input_string)]
    )


def main():
    with open("../files/2017_01_input.txt") as f:
        input_string = f.read().strip()

    print(part1(input_string))
    print(part2(input_string))


if __name__ == "__main__":
    main()
